src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -14,14 +14,46 @@ func (st *SectorTable_I) InactivePower() block.StoragePower {
 	return block.StoragePower((st.CommittedSectors_ + st.RecoveringSectors_ + st.FailingSectors_) * util.UVarint(st.SectorSize_))
 }
 
+func (cs *ChallengeStatus_I) OnNewChallenge(currEpoch block.ChainEpoch) ChallengeStatus {
+	cs.LastChallengeEpoch_ = currEpoch
+	return cs
+}
+
+// Call by either SubmitPoSt or OnMissedPoSt
+// TODO: verify this is correct and if we need to distinguish SubmitPoSt vs OnMissedPoSt
+func (cs *ChallengeStatus_I) OnChallengeResponse(currEpoch block.ChainEpoch) ChallengeStatus {
+	cs.LastResponseEpoch_ = currEpoch
+	return cs
+}
+
+func (cs *ChallengeStatus_I) isChallenged() bool {
+	// true (isChallenged) when LastChallengeEpoch is later than LastResponseEpoch
+	return cs.LastChallengeEpoch() > cs.LastResponseEpoch()
+}
+
 // called by CronActor to notify StorageMiner of PoSt Challenge
 func (sm *StorageMinerActor_I) NotifyOfPoStChallenge() {
-	if sm.isChallenged() {
-		// TODO: proper throw
-		panic("should not be challenged when isChallenged")
+
+	if sm.ChallengeStatus().Impl().isChallenged() {
+		return // silent return, dont re-challenge
+	}
+
+	// TODO: get currEpoch from runtime
+	var currEpoch block.ChainEpoch
+	sm.ChallengeStatus().Impl().OnNewChallenge(currEpoch)
+
+	// TODO: commit state change
+}
+
+func (sm *StorageMinerActor_I) loadStagedCommittedSectors() {
```

Comment:
`load` is an odd name for this (it's fine -- bigger fish to fry), but if we change other things, probably something like `processStagedCommittedSectors`

Reply:
this turned out pretty well, good solution 👍